sympy/physics/vector/printing.py

```diff
@@ -53,70 +53,18 @@ def _print_Function(self, expr, exp=None):
             not isinstance(type(expr), UndefinedFunction):
             return getattr(self, '_print_' + func)(expr, exp)
         elif isinstance(type(expr), UndefinedFunction) and (expr.args == (t,)):
-
-            name, supers, subs = split_super_sub(func)
-            name = translate(name)
-            supers = [translate(sup) for sup in supers]
-            subs = [translate(sub) for sub in subs]
-
-            if len(supers) != 0:
-                supers = r"^{%s}" % "".join(supers)
-            else:
-                supers = r""
-
-            if len(subs) != 0:
-                subs = r"_{%s}" % "".join(subs)
+            # treat this function like a symbol
+            expr = Symbol(func)
+            if exp is not None:
+                # copied from LatexPrinter._helper_print_standard_power, which
+                # we can't call because we only have exp as a string.
+                base = self.parenthesize(expr, PRECEDENCE['Pow'])
+                base = self.parenthesize_super(base)
+                return r"%s^{%s}" % (base, exp)
             else:
-                subs = r""
-
-            if exp:
-                supers += r"^{%s}" % exp
-
-            return r"%s" % (name + supers + subs)
+                return super()._print(expr)
```

Comment:
This change means that `self._settings['symbol_names']` now works for dynamics symbols.